robottelo/report_portal/portal.py

```diff
@@ -0,0 +1,204 @@
+
+        :returns dict: parameters for API launch request
+        """
+        _params = {
+            'page.page': 1,
+            'page.size': 500,
+            'page.sort': 'start_time',
+        }
+        return _params
+
+    @property
+    def launches(self):
+        """Returns All launches in Report Portal
+
+        This does not includes each tests data for all tests in all launches.
+        But it includes just an overview data for all and each launch in Report Portal
+
+        :returns list: List of all launches in Report Portal
+        """
+        resp = requests.get(
+            self.launch_url, headers=self.headers, params=self.params, verify=False
+        )
+        resp.raise_for_status()
+        data = resp.json()
+        if data['content']:
+            launches = sorted(data['content'], key=lambda x: x['start_time'], reverse=True)
+            return launches
+
+    def launch(self, latest):
+        """Returns a specific launch overview data
+
+        This does not includes each tests data in launch
+
+        :param int latest: The launch place in Report portal
+            e.g To get the latest launch in Report Portal, ser latest = 1,
+                For second latest, set latest = 2
+
+        :returns dict: The launch overview data of requested launch
+        """
+        latest = int(latest) if isinstance(latest, str) else latest
+        return self.launches[int(latest) - 1]
+
+
+class Tests(Launch):
+    """Represents Tests in a Launch of a Report Portal"""
+
+    def __init__(self, launch):
+        """Initiates launch and properties for tests in that launch"""
+        super(Tests, self).__init__()
+        self.tests_url = f'{self.api_url}/item'
+        self.launch = self.launch(latest=launch)
+
+    @property
+    def defect_types(self):
+        """The various defect types in Report Report Portal for tests
+
+        This holds the codes for all the defect types, the tests will be marked with.
+
+        :returns dict: The dict of varios defect types and their codes
+        """
+        _defect_types = {
+            'product_bug': 'PB001',
+            'automation_bug': 'AB001',
+            'system_issue': 'SI001',
+            'to_investigate': 'TI001',
+            'no_defect': 'ND001',
+        }
+        return _defect_types
+
+    def _params(self, **kwargs):
+        """Parameters for Test items API request
+
+        :param dict kwargs: The tests `defect_type` and/or testss `status`
+
+        It contains filters to get the test items data from launch
+
+        :returns dict: parameters for API test items request
+        """
+        params = [
+            ('filter.eq.launch', self.launch['id']),
+            ('page.page', 1),
+            ('page.size', 50),
+            ('page.sort', 'start_time'),
+        ]
+        defect_type = kwargs.get('defect_type')
+        test_status = kwargs.get('status')
+        if defect_type and defect_type in self.defect_types:
```

Comment:
if defect_type and defect_type in self.defect_types:
```suggestion
        if defect_type in self.defect_types:
```